### PR TITLE
Update search geography geocode

### PR DESCRIPTION
**Pull request by shinson**
Update the spending by geography filter to include county and congressional districts as well as map filter. 

Story: DS-1716 & DS-1719

- [x] Frontend Approved Gist : https://gist.github.com/shinson/3f67d9ac3d72d66e295c7f978a471986
- [x]  Endpoint matches Gist
- Moving improving performance testing to next sprint

### CODE REVIEW

File: usaspending_api/search/v2/views/search.py [hunk 1]
```diff
@@ -460,64 +463,169 @@ def post(self, request):
...
+        self.geo_queryset = self.queryset.filter(**filter_args) \
+            .values(*lookup_fields) \
+            .annotate(federal_action_obligation=Sum('federal_action_obligation'))
+
+        # State names are inconsistent in database (upper, lower, null)
+        # Used lookup instead to be consistent
+        results = [
+            {
+                'shape_code': x[loc_lookup],
+                'federal_action_obligation': x['federal_action_obligation'],
```
[the remark this answers is not in the document]
Comment (shinson) [reply usaspending_api/search/v2/views/search.py]: Updated to 'aggregated_amount' for state, county, district

File: usaspending_api/search/v2/views/search.py [hunk 2]
```diff
@@ -18,11 +19,13 @@
...
 from usaspending_api.awards.v2.filters.transaction import transaction_filter
 from usaspending_api.awards.v2.filters.award import award_filter
+from usaspending_api.awards.v2.filters.location_filter_geocode import geocode_filter_locations
 from usaspending_api.awards.v2.lookups.lookups import award_contracts_mapping, contract_type_mapping, \
     loan_type_mapping, loan_award_mapping, non_loan_assistance_award_mapping, non_loan_assistance_type_mapping
+from usaspending_api.references.abbreviations import code_to_state, code_to_fips, fips_to_code, pad_codes
 from usaspending_api.references.models import Cfda, LegalEntity
 
-
+import timeit
```
[the remark this answers is not in the document]
Comment (shinson) [reply usaspending_api/search/v2/views/search.py]: removed

File: usaspending_api/search/v2/views/search.py [hunk 3]
```diff
@@ -460,64 +462,171 @@ def post(self, request):
...
+        # need to split for the geocode_filter
+        if self.geo_layer_filters:
+            self.queryset &= geocode_filter_locations(scope_field_name, [
+                {'state': fips_to_code.get(x[:2]), self.geo_layer: x[2:], 'country': 'USA'}
+                for x in self.geo_layer_filters
+            ], 'TransactionNormalized')
+
+        # Turn county/district codes into float since inconsistent in database
+        # Codes in location table ex: '01', '1', '1.0'
+        # Cast will group codes as a float and will combine inconsistent codes
```
Comment (tony-sappe) [inline usaspending_api/search/v2/views/search.py]: Good comment

File: usaspending_api/references/abbreviations.py [hunk 4]
```diff
@@ -1,66 +1,67 @@
...
+    'OR': {'name': 'OREGON', 'fips': '41'},
+    'PA': {'name': 'PENNSYLVANIA', 'fips': '42'},
+    'PR': {'name': 'PUERTO RICO', 'fips': '72'},
+    'PW': {'name': 'PALAU', 'fips': '70'},
+    'RI': {'name': 'RHODE ISLAND', 'fips': '44'},
+    'SC': {'name': 'SOUTH CAROLINA', 'fips': '45'},
+    'SD': {'name': 'SOUTH DAKOTA', 'fips': '46'},
+    'TN': {'name': 'TENNESSEE', 'fips': '47'},
+    'TX': {'name': 'TEXAS', 'fips': '48'},
+    'UM': {'name': 'UTAH', 'fips': '49'},
```
Comment (tony-sappe) [inline usaspending_api/references/abbreviations.py]: Looks like a typo, lines 53 and 54

Comment (shinson) [reply usaspending_api/references/abbreviations.py]: fixed

**Pull request merged**